homeassistant/components/speedtestdotnet/sensor.py

```diff
@@ -69,47 +82,27 @@ def should_poll(self):
     @property
     def device_state_attributes(self):
         """Return the state attributes."""
-        attributes = {ATTR_ATTRIBUTION: ATTRIBUTION}
-        if self._data is not None:
-            return attributes.update(
-                {
-                    ATTR_BYTES_RECEIVED: self._data["bytes_received"],
-                    ATTR_BYTES_SENT: self._data["bytes_sent"],
-                    ATTR_SERVER_COUNTRY: self._data["server"]["country"],
-                    ATTR_SERVER_ID: self._data["server"]["id"],
-                    ATTR_SERVER_LATENCY: self._data["server"]["latency"],
-                    ATTR_SERVER_NAME: self._data["server"]["name"],
-                }
-            )
+        attributes = {
+            ATTR_ATTRIBUTION: ATTRIBUTION,
+            ATTR_SERVER_NAME: self.coordinator.data["server"]["name"],
+            ATTR_SERVER_COUNTRY: self.coordinator.data["server"]["country"],
+            ATTR_SERVER_ID: self.coordinator.data["server"]["id"],
+        }
+        if self.type == "download":
+            attributes[ATTR_BYTES_RECEIVED] = self.coordinator.data["bytes_received"]
+
+        if self.type == "upload":
+            attributes[ATTR_BYTES_SENT] = self.coordinator.data["bytes_sent"]
+
         return attributes
 
     async def async_added_to_hass(self):
         """Handle entity which will be added."""
-        await super().async_added_to_hass()
-        state = await self.async_get_last_state()
```

Comment:
Should we not keep the restored state feature?

Reply:
I am not sure if that has any benefit as the coordinator will update the data on startup. If it fails `ConfigEntryNotReady` will be raised.

Reply:
Right, ok.